pkg/gitutils.TestCheckoutGit fails depending on gitconfig core.autocrlf

**Steps to reproduce the issue:**
```
$ git config --get core.autocrlf
$ go test ./pkg/gitutils
```
Tested with 4a7715c9783627ecc194dac8502a684f40a9d0bd

**Describe the results you received:**

OS|core.autocrlf|Result
---|---|---
Linux|true|FAIL
Linux|false/unset|PASS
Windows|true|PASS
Windows|false/unset|FAIL

failure message:
```
--- FAIL: TestCheckoutGit (0.04s)
        gitutils_test.go:201: Expected FROM scratch
                EXPOSE 5000, was FROM scratch
                EXPOSE 5000
FAIL
```


**Describe the results you expected:**
It should not fail.

**Additional information you deem important (e.g. issue happens only occasionally)**
* Found in PR #22930, which is an attempt to simplify go test in hack/*. The PR uses `env -i` for `test-unit` as well as for `test-integration`, so it disables some git-related environment variable.

* The cause of the issue is obvious: `eol` is not properly set : https://github.com/docker/docker/blob/4a7715c9783627ecc194dac8502a684f40a9d0bd/pkg/gitutils/gitutils_test.go#L80

--- a/pkg/gitutils/gitutils_test.go
+++ b/pkg/gitutils/gitutils_test.go
@@ -10,6 +10,7 @@ import (
 	"path/filepath"
 	"reflect"
 	"runtime"
+	"strings"
 	"testing"
 )
 
@@ -70,6 +71,16 @@ func TestCloneArgsStripFragment(t *testing.T) {
 	}
 }
 
+func gitGetConfig(name string) string {
+	b, err := git([]string{"config", "--get", name}...)
+	if err != nil {
+		// since we are interested in empty or non empty string,
+		// we can safely ignore the err here.
+		return ""
+	}
+	return strings.TrimSpace(string(b))
+}
+
 func TestCheckoutGit(t *testing.T) {
 	root, err := ioutil.TempDir("", "docker-build-git-checkout")
 	if err != nil {
@@ -77,8 +88,13 @@ func TestCheckoutGit(t *testing.T) {
 	}
 	defer os.RemoveAll(root)
 
+	autocrlf := gitGetConfig("core.autocrlf")
+	if !(autocrlf == "true" || autocrlf == "false" ||
+		autocrlf == "input" || autocrlf == "") {
+		t.Logf("unknown core.autocrlf value: \"%s\"", autocrlf)
+	}
 	eol := "\n"
-	if runtime.GOOS == "windows" {
+	if autocrlf == "true" {
 		eol = "\r\n"
 	}
 

